### PR TITLE
Remove Field and Add Config Mapping

### PR DESCRIPTION
**Pull request by jgbenito7**
Removes the need to have `Field` components as wrappers for inputs and instead uses a config file to know how to apply props to a given field.

### CODE REVIEW

File: modules/mapField.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import R from 'ramda'
+
+const isEvent = e => R.has('nativeEvent', e)
```
Comment (beardedtim) [inline modules/mapField.js]: I was silly here. Can we change this to

IsEvent = R.has('nativeEvent')

?

File: modules/mapField.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        updateError(e)
+      }
+
+      const value = getValueFromEvent(e)
+
+      return updateFormValueForKey(value)
+    }
+
+    const validateEvent = {
+      focus: 'onFocus',
```
Comment (beardedtim) [inline modules/mapField.js]: Does this need to be defined inside this scope?

**Comment by beardedtim** [comment]
Can we also get a doc update for the proposed usage?

File: modules/mapField.js (hunk 1 above)
Comment (jgbenito7) [reply modules/mapField.js]: Ha.. yeah

File: README.md [hunk 3]
```diff
@@ -26,44 +30,16 @@ class MyForm extends React.Component {
...
-              const { error } = getErrorProps()
-
-              return (
-                <label htmlFor={htmlFor}>Your location Is?</label>
-                <textarea id={id} value={value} {...props} />
-              )
-            }}
-          </Form.Field>
-        </Form>
+        <Form onSubmit={this.handleSubmit} config={config} render={(fields) => (
```
Comment (beardedtim) [inline README.md]: Yassssss! Now I can not ask you guys how this works tomorrow!

**Comment by jgbenito7** [comment]
> Can we also get a doc update for the proposed usage?

done

File: README.md [hunk 4]
```diff
@@ -4,19 +4,23 @@
...
+import { Form } from '@kofile/react-form'
+
+const config = {
+  email: {
+    validateOn: 'change',
+    validateWith: v => v ? '' : 'error message'
+  },
+  password: {
+    validateOn: 'blur',
+    validateWith: v => v ? '' : 'fix your password!!!'
```
Comment (neezer) [inline README.md]: I really feel both `validateWith` and `validateOn` should be arrays.

File: README.md [hunk 5]
```diff
@@ -4,19 +4,23 @@
...
   }
+}
 
-  handleSubmit = ({ formValues, formErrors, isValid }) => {
+class MyForm extends React.Component {
+  handleSubmit = ({ formValues, isValid }) => {
     if (!isValid) {
-      return handleErrors(formErrors)
+      return;
```
Comment (neezer) [inline README.md]: `prettier-standard` didn't format this ... ?

File: README.md [hunk 6]
```diff
@@ -26,44 +30,16 @@ class MyForm extends React.Component {
...
-              return (
-                <label htmlFor={htmlFor}>Your location Is?</label>
-                <textarea id={id} value={value} {...props} />
-              )
-            }}
-          </Form.Field>
-        </Form>
+        <Form onSubmit={this.handleSubmit} config={config} render={(fields) => (
+          <div>
+            <label {...fields.email.labelProps}>Email</label>
```
Comment (neezer) [inline README.md]: `htmlFor`'s value is just an id, more or less. Maybe just use

```js
fields.email = Object.assign('email', { inputProps: emailInputProps })
```

Then you could forgo `labelProps` altogether and just do

```js
<label htmlFor={fields.email}>Email</label>
<input {...fields.email.inputProps} />
```

Additionally, should you want to write labels/inputs like so:

```js
<label>
  <span>Email</span>
  <input {...fields.email.inputProps} />
</label>
```

... you won't be left with a useless prop grouping.

File: README.md (hunk 5 above)
Comment (jgbenito7) [reply README.md]: It's a README, so probably doesn't do that

Comment (beardedtim) [reply README.md]: It might have and decided it needed the semi colon here due to the empty `return`?

File: README.md (hunk 4 above)
Comment (jgbenito7) [reply README.md]: I wanted to get things working how they were first with the new format, then add things like multiple validations and async validations

File: README.md (hunk 6 above)
Comment (jgbenito7) [reply README.md]: are we 100% sure that label will only ever need one prop though?

**Comment by jgbenito7** [comment]
> Did index.js get updated? Only Form should be exported publically.

Yes it did

**Pull request merged**